export/export/export.go

```diff
@@ -0,0 +1,408 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package export provides helpers for exporting pipelines to BigQuery.
+package export
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	genomicsv1 "google.golang.org/api/genomics/v1alpha2"
+	genomics "google.golang.org/api/genomics/v2alpha1"
+	"google.golang.org/api/iterator"
+)
+
+type Config struct {
+	Project, Filter, Dataset, Table string
+	Update                          bool
+}
+
+func (c *Config) ExportV2(ctx context.Context, service *genomics.Service) error {
```

Comment:
Can we not simply have support for exporting to bigquery and leave the per version bits in their respective places?

It looks a bit like you need something like an interface than can page through data and return a struct that gets encoded by the bigquery exporter?  Pivoting the design like that should keep the common code here and the v1/v2 specific code where it belongs...

Reply:
Moved the version specific stuff.